api/v1alpha1/bundle_types.go

```diff
@@ -40,9 +42,20 @@ const (
 type BundleSpec struct {
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
 	ProvisionerClassName string `json:"provisionerClassName"`
+	// Source defines the configuration for the underlying Bundle content.
+	Source BundleSource `json:"source"`
+}
 
+type BundleSource struct {
+	// Type defines the kind of Bundle content being sourced.
+	Type string `json:"type"`
 	// Image is the bundle image that backs the content of this bundle.
-	Image string `json:"image"`
+	Image *ImageSource `json:"image,omitempty"`
+}
+
+type ImageSource struct {
```

Comment:
I think we have to update the corresponding printer column to reflect the new spec field on the API
```
//+kubebuilder:printcolumn:name=Image,type=string,JSONPath=`.spec.source.image.ref`
```

See https://github.com/operator-framework/rukpak/issues/163 for tracking issue on generalizing the field.